### ISSUE TITLE
DistributionTest/GammaDistributionFittingTest failed 

### ISSUE THREAD

**Issue by dongli** [mlpack/mlpack#911]
I built `mlpack` 2.1.1 on macOS 10.12.3 with clang-800.0.42.1. `mlpack` is built successfully, but when test it failed:
```
mlpack-2.1.1/src/mlpack/tests/distribution_test.cpp:481: fatal error: in "DistributionTest/GammaDistributionFittingTest": difference{0.11739} between gDist.Beta(0){1.6760847564140695} and betaReal{1.5} exceeds 10%
```

**Comment by mentekid** [comment]
Hi Dong,

This is probably a random error. Some times the fitting does not converge to the right number.

Can you try running the test again? We might need to loosen the tolerance a little more...

**Comment by dongli** [comment]
@mentekid I have tried several times with no luck.

**Comment by zoq** [comment]
Can you confirm that you rebuild the test every time or used another seed for every run something by using something like:

``math::RandomSeed((size_t) std::time(NULL));``

**Comment by rcurtin** [comment]
Hey Li,

Sorry about this.  I actually ended up fixing this 5 days after 2.1.1 was released but since there has not been a new version released yet, there is not a new version for you to update to.  Like @mentekid said, this was a random failure caused by variance in the tests.  0b5d5f1c was the commit that reduced the variance in the tests.

2.2.0 will be released soon (I hope this week, or really, I hope tomorrow, since it doesn't look like it'll happen today...) and this will take care of the issue.  Random test failures are hard to keep down with probabilistic tests...
